opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -189,6 +231,36 @@ public function index() {
         $this->response->setOutput($this->render());
     }
 
+    protected function handleLayoutSaving() {
+        if (!$this->user->hasPermission('modify', 'module/smaily_for_opencart')) {
+            return;
+        }
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Check if layout settings are in POST & that it's an array.
+        if (!empty($this->request->post['smaily_for_opencart_module']) && is_array($this->request->post['smaily_for_opencart_module'])) {
+            $smaily_modules_san = [];
+            // Declare layout module settings in POST.
+            $smaily_modules = $this->request->post['smaily_for_opencart_module'];
```

Comment:
Don't use brand names in variable names. Having `$modules` is totally fine, and it does not conflict with any of the variables used here.

Also you are not checking if posted data is an array.

Reply:
Ok, will keep it in mind.
The last sentence, doesn't `is_array` accomplish this or have I misunderstood it?

Reply:
Yes, you are right. I did not notice the `is_array` on line 241.